src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,1910 @@
+        seed_number (int, *optional*):
+            Value to set for the random seed.
+
+    Returns:
+        `tuple(torch.Tensor)`: inputs_mask, masked input, same shape as input Tensor and mask tensor of shape [bs x c x
+        n]
+    """
+    if seed_number:
+        set_seed(seed_number)
+
+    batch_size, num_channels, sequence_length, num_features = inputs.shape
+    device = inputs.device
+
+    len_keep = int(sequence_length * (1 - mask_ratio))
+
+    if channel_consistent_masking:
+        noise = torch.rand(batch_size, 1, sequence_length, device=device)  # noise in [0, 1], bs x 1 x  L
+        noise = noise.repeat(1, num_channels, 1)  # bs x num_channels x time
+    else:
+        noise = torch.rand(
+            batch_size, num_channels, sequence_length, device=device
+        )  # noise in [0, 1], bs x num_channels x L
+
+    mask = torch.ones(
+        batch_size, num_channels, sequence_length, device=device
+    )  # mask: [bs x num_channels x num_patch]
+    mask[:, :, :len_keep] = 0
+
+    # sort noise for each sample
+    ids_shuffle = torch.argsort(noise, dim=-1)  # ascend: small is keep, large is remove
+    ids_restore = torch.argsort(ids_shuffle, dim=-1)  # ids_restore: [bs x num_channels x L]
+
+    mask = torch.gather(mask, dim=-1, index=ids_restore)
+    mask = mask.unsqueeze(-1).repeat(1, 1, 1, num_features)  # mask: [bs x num_channels x num_patches x patch_length]
+    if unmasked_channel_indices is not None:
+        mask[:, unmasked_channel_indices, :, :] = 0
+
+    inputs_mask = inputs.masked_fill(mask.bool(), mask_value)
+    return inputs_mask, mask[..., 0]
+
+
+def forecast_masking(
+    inputs: torch.Tensor,
+    forecast_mask_patches: list,
+    forecast_mask_ratios: list = None,
+    unmasked_channel_indices: list = None,
+    mask_value: int = 0,
+    seed_number: Optional[int] = None,
+):
+    """Forecast masking that masks the last K patches where K is from the forecast_mask_patches list.
+    For every batch, distribute the patch lengths based on forecast_mask_ratios and ignore masks for column indices
+    mentioned in unmasked_channel_indices.
+
+    Parameters:
+        inputs (`torch.Tensor`):
+            Input of shape `(bs, num_channels, num_patch, patch_len)` or `(bs, tsg1, tag2, num_channels, num_patch,
+            patch_len)`
+        forecast_mask_patches (`list`): [2, 4]
+            List of patch lengths to mask in the end of the data.
+        forecast_mask_ratios (`list`, *optional*): [0.7, 0.3]
+            List of weights to use for each patch length. For Ex. if forecast_mask_patches is [5,4] and
+            forecast_mask_ratios is [1,1], then equal weights to both patch lengths. Defaults to None.
+        unmasked_channel_indices (`list`, *optional*):
+            Control Variable channel indices. These channels will not be masked. Defaults to None.
+        mask_value (`int`, *optional* defaults to 0):
+            Value to use for masking. Defaults to 0.
```

Comment:
Already in the docstring info 

```suggestion
            Value to use for masking.
```